src/main/java/at/ac/tuwien/kr/alpha/grounder/NaiveGrounder.java

```diff
@@ -296,7 +311,11 @@ public AnswerSet assignmentToAnswerSet(Iterable<Integer> trueAtoms) {
 		maxAtomIdBeforeGroundingNewNoGoods = atomStore.getMaxAtomId();
 		// Compute new ground rule (evaluate joins with newly changed atoms)
 		for (IndexedInstanceStorage modifiedWorkingMemory : workingMemory.modified()) {
-			if (modifiedWorkingMemory.getPredicate().isInternal()) {
+			// Skip predicates solely used in the solver which do not occur in rules.
+			Predicate workingMemoryPredicate = modifiedWorkingMemory.getPredicate();
+			if (workingMemoryPredicate == RuleAtom.PREDICATE
```

Comment:
This easily produces errors when new internal predicates are introduced. Why aren´t we using `getPredicate().isInternal()` anymore?